pkg/autodiscovery/listeners/cloudfoundry_test.go

```diff
@@ -389,7 +392,12 @@ func TestCloudFoundryListener(t *testing.T) {
 		testBBSCache.DesiredLRPs = tc.dLRP
 		testBBSCache.Unlock()
 
-		time.Sleep(50 * time.Millisecond)
+		testutil.RequireTrueBeforeTimeout(t, 15*time.Millisecond, 250*time.Millisecond, func() bool {
+			// wait until at least one listener refresh loop passes
+			return atomic.LoadInt64(&cfl.refreshCount) > lastRefreshCount
```

Comment:
is this the right condition to wait on? Unless I'm missing something, this tells us whether a refresh has started, not whether it's finished.

Reply:
Ouch, you're right, I'll just fix this using the lock that already exists on `cfl`.